core/optaplanner-core-impl/src/main/java/org/optaplanner/core/api/solver/change/ProblemChangeDirector.java

```diff
@@ -34,66 +34,78 @@
     /**
      * Add a new {@link PlanningEntity} instance into the {@link PlanningSolution working solution}.
      *
+     * @deprecated as not necessary for implementing {@link ProblemChange}s.
      * @param entity never null; the {@link PlanningEntity} instance
      * @param entityConsumer never null; adds the entity to the {@link PlanningSolution working solution}
      * @param <Entity> the planning entity object type
      */
+    @Deprecated(forRemoval = true)
     <Entity> void addEntity(Entity entity, Consumer<Entity> entityConsumer);
```

Comment:
Also, if they are not necessary, why not make them default?
That way, new users will likely never get to see them.